sycl/test-e2e/Reduction/reduction_initialize_to_identity.cpp

```diff
@@ -0,0 +1,41 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <cstdio>
+#include <cstdlib>
+#include <numeric>
+
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/oneapi/experimental/reduction_properties.hpp>
+
+namespace syclex = sycl::ext::oneapi::experimental;
+
+int sum(sycl::queue q, int *input, size_t N) {
+
+  int result = 42;
+  {
+    sycl::buffer<int> buf{&result, 1};
+
+    q.submit([&](sycl::handler &h) {
+      auto reduction =
+          sycl::reduction(buf, h, sycl::plus<>(),
+                          syclex::properties(syclex::initialize_to_identity));
+      h.parallel_for(N, reduction,
+                     [=](size_t i, auto &reducer) { reducer += input[i]; });
+    });
+  }
+  return result;
+}
+
+int main(int argc, char *argv[]) {
+
+  constexpr size_t N = 32;
+  int *array = new int[N];
```

Comment:
Why heap and not a normal array/`std::array`?

Reply:
Force of habit.  I rewrote things to use a `std::array` in c39e14f18f5535ad8b1afb8a105c3c5d4d93df5f.